common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -497,4 +503,44 @@ public static String hashString(String input) {
         }
     }
 
+    /**
+     * Checks if the input is safe (non-null, non-blank, matches safe character set).
+     *
+     * @param value The input string to validate
+     * @return true if input is safe, false otherwise
+     */
+    public static boolean isSafeText(String value) {
+        if (value == null || value.isBlank()) {
+            return false;
+        }
+        return value.matches(SAFE_INPUT_REGEX);
```

Comment:
Pattern SAFE_TEXT_PATTERN = Pattern.compile(SAFE_INPUT_REGEX); Using Pattern matcher is more efficient as it only Compile pattern once rather than Compiles pattern every time in String.matches():

Reply:
Agree, addressed in the next commit.